java/util/PriorityQueue/PriorityQueue.hpp

```diff
@@ -22,4 +22,274 @@
  * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
  * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
- */
\ No newline at end of file
+ */
+
+#ifndef JAVA_UTIL_PRIORITY_QUEUE_HPP
+#define JAVA_UTIL_PRIORITY_QUEUE_HPP
+
+#include "../Collection/Collection.hpp"
+#include "../Comparator/Comparator.hpp"
+#include "../../lang/Exception/Exception.hpp"
+#include "../../lang/ClassCastException/ClassCastException.hpp"
+#include <vector>
+
+namespace Java {
+    namespace Util {
+        template <typename E>
+        class PriorityQueue
+//      :
+//      public Serializable
+//      public AbstractQueue<E>
+        {
+        private:
+            std::vector<E> original;
+
+        public:
+            /**
+             * Constructor
+             * Creates a empty PriorityQueue.
+             */
+            PriorityQueue() { };
+
+            /**
+             * Constructor
+             * Creates a PriorityQueue containing the elements in the specified collection.
+             *
+             * @param target
+             */
+//          PriorityQueue(const Collection<E> &target);
+
+            /**
+             * Constructor
+             * Creates a PriorityQueue whose elements are ordered according to the specified comparator.
+             *
+             * @param target
+             */
+//          PriorityQueue(const Comparator<E> &target);
+
+            /**
+             * Constructor
+             * Creates a PriorityQueue with the specified initial capacity.
+             *
+             * @param initialCapacity
+             */
+//          PriorityQueue(int initialCapacity);

```

Comment:
I have enabled it, but I can only test this constructor with `initialCapacity < 1` to throw exception. PriorityQueue class doesn't have "getCapacity" (or similar) method and std::vector<T> doesn't have any method to control exactly initialCapacity.
We use this constructor to ensure that PriorityQueue instance has "enough capacity" (>= initialCapacity argument) as we want.